vpp-manager/vpp_runner.go

```diff
@@ -53,49 +53,75 @@ type VppRunner struct {
 	params       *config.VppManagerParams
 	conf         *config.InterfaceConfig
 	vpp          *vpplink.VppLink
-	uplinkDriver uplink.UplinkDriver
+	uplinkDriver []uplink.UplinkDriver
 	routeWatcher *RouteWatcher
 	poolWatcher  *PoolWatcher
 	linkWatcher  *LinkWatcher
 }
 
-func NewVPPRunner(params *config.VppManagerParams, conf *config.InterfaceConfig) *VppRunner {
+func NewVPPRunner(params *config.VppManagerParams, confs *config.InterfaceConfig) *VppRunner {
 	return &VppRunner{
 		params: params,
-		conf:   conf,
+		conf:   confs,
 	}
 }
 
-func (v *VppRunner) Run(driver uplink.UplinkDriver) error {
-	v.uplinkDriver = driver
-	log.Infof("Running with uplink %s", driver.GetName())
+func (v *VppRunner) GenerateVppConfigExecFile() error {
+	template := config.TemplateScriptReplace(v.params.ConfigExecTemplate, v.params, v.conf)
```

Comment:
It looks like you went through the trouble of modifying the functions handling the config files in the drivers but aren't calling them anywhere - I think thins function and the next one (GenerateVppConfigFile) are the right place to do so.